hash_practice/exercises.py

```diff
@@ -2,18 +2,40 @@
 def grouped_anagrams(strings):
     """ This method will return an array of arrays.
         Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n)
+        Space Complexity: O(n)
     """
-    pass
+    anagrams_map = {}
+    for word in strings:
+        sorted_word = "".join(sorted(word))
+        if sorted_word not in anagrams_map:
+            anagrams_map[sorted_word] = [word]
+        else:
+            anagrams_map[sorted_word].append(word)
+    return list(anagrams_map.values())
+
 
 def top_k_frequent_elements(nums, k):
     """ This method will return the k most common elements
         In the case of a tie it will select the first occuring element.
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n)
+        Space Complexity: O(n)
     """

```

Comment:
👍 Good use of the `max` function!

Given that I would say that this is O(nk) given the loop and `max` function.